pyqtorch/modules/parametric.py

```diff
@@ -9,11 +9,11 @@
     create_controlled_batch_from_operation,
 )
 from pyqtorch.core.utils import OPERATIONS_DICT
+from pyqtorch.modules import QuantumCircuit
+from pyqtorch.modules.utils import rot_matrices
 
 
-class RotationGate(Module):
-    n_params = 1
-
+class Gate(Module):
```

Comment:
you are kind of addressing #46 here as well;) depending on how much work you want to do here you could give the `AbstractGate` idea a shot to properly formalize this class hierarchy we have in pyq. But then you will likely have to touch almost all of the gates. Primitive/rotation/controlled etc. so I think it should be in a separate PR.

Reply:
I think I'll start here since it would make annotations much more compact.